src/Operation/Count.php

```diff
@@ -187,7 +187,7 @@ private function createCommandDocument(): array
     {
         $cmd = ['count' => $this->collectionName];
 
-        if (! empty($this->filter)) {
+        if ($this->filter !== []) {
```

Comment:
> `$filter` is an array or an object and even `empty(new stdClass)` is true. This might be a necessary bug fix.

The [count](https://www.mongodb.com/docs/manual/reference/command/count/) and [distinct](https://www.mongodb.com/docs/manual/reference/command/distinct/) commands both use an empty filter by default, so including `new stdClass` in the outgoing command was inconsistent (and redundant) but had no functional impact on the result.

I don't think we need to go the extra mile to detect empty objects, so if this is the most concise change to silence Psalm that works for me.